src/Object/Contact.php

```diff
@@ -469,49 +460,31 @@ public static function photoMenu(array $contact, $uid = 0)
 	}
 
 	/**
+	 * @brief Returns ungrouped contact count or list for user
+	 *
 	 * Returns either the total number of ungrouped contacts for the given user
 	 * id or a paginated list of ungrouped contacts.
 	 *
-	 * @brief Returns ungrouped contact count or list for user
+	 * @param int $uid   uid
+	 * @param int $start optional, default 0
+	 * @param int $count optional, default 0
 	 *
-	 * @param int $uid
-	 * @param int $start
-	 * @param int $count
 	 * @return array
 	 */
 	public static function getUngroupedList($uid, $start = 0, $count = 0)
 	{
 		if (!$count) {
-			$r = q(
-				"SELECT COUNT(*) AS `total`
-				 FROM `contact`
-				 WHERE `uid` = %d
-				 AND `self` = 0
-				 AND `id` NOT IN (
-					SELECT DISTINCT(`contact-id`)
-					FROM `group_member`
-					WHERE `uid` = %d
-				) ", intval($uid), intval($uid)
-			);
+			$fields = array('COUNT(*) AS `total`');
+			$condition = array('`uid` = ? AND `self` = 0 AND `id` NOT IN (SELECT DISTINCT(`contact-id`)	FROM `group_member`	WHERE `uid` = ?', $uid, $uid);
+			$r = dba::select('contact', $fields, $condition);
 
-			return $r;
-		}
-
-		$r = q(
-			"SELECT *
-			FROM `contact`
-			WHERE `uid` = %d
-			AND `self` = 0
-			AND `id` NOT IN (
-				SELECT DISTINCT(`contact-id`)
-				FROM `group_member` WHERE `uid` = %d
-			)
-			AND `blocked` = 0
-			AND `pending` = 0
-			LIMIT %d, %d", intval($uid), intval($uid), intval($start), intval($count)
-		);
+			return dba::inArray($r);
+		}
 
-		return $r;
+		$innerCondition = array('`id` NOT IN (SELECT DISTINCT(`contact-id`) FROM `group_member` WHERE `uid` = ?', $uid);
+		$r = dba::select('contact', array(), array('uid' => $uid, 'self' => 0, $innerCondition, 'blocked' => 0, 'pending' => 0), array('limit ?, ?', $start, $count));
```

Comment:
I don't think that the condition will work that way. Please use ```dba::p``` instead. Same is valid for the ```limit```.

I realised some ugly behaviour with prepared statements: The values for ```LIMIT``` can't be transmitted via ```?```, because this way they are transmitted as string and a ```LIMIT '1'``` creates an error. This is some limitation of PDO and MySQLi.